static/src/js/components/ProjectPanels/ProjectPanels.js

```diff
@@ -22,203 +25,392 @@ import { isAccessMethodValid } from '../../util/accessMethods'
-
-  const { collectionIds: projectIds, byId: projectById } = project
-
-  const { activePanel, isOpen } = projectPanels
-  const panelSectionEditOptions = []
-  const panelSectionCollectionDetails = []
-
-  let loaded = false
-  if (projectIds[0] && !Object.keys(byId[projectIds[0]].metadata).length) return null
-
-  const onPanelClose = () => {
+
+class ProjectPanels extends PureComponent {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      selectedKeyword: null,
+      selectedVariable: null,
+      selectedVariables: {},
+      variables: null
+    }
+
+    this.onPanelClose = this.onPanelClose.bind(this)
+    this.onChangePanel = this.onChangePanel.bind(this)
+    this.selectKeyword = this.selectKeyword.bind(this)
+    this.onSaveVariables = this.onSaveVariables.bind(this)
+    this.onCheckboxChange = this.onCheckboxChange.bind(this)
+    this.clearSelectedKeyword = this.clearSelectedKeyword.bind(this)
+    this.clearSelectedVariable = this.clearSelectedVariable.bind(this)
+    this.onViewDetails = this.onViewDetails.bind(this)
+    this.backToOptions = this.backToOptions.bind(this)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // const { selectedVariables: oldSelectedVariables } = this.state
+    const { project: nextProject } = nextProps
+    const { byId, collectionIds } = nextProject
+
+    const selectedVariables = {}
+
+    collectionIds.forEach((collectionId) => {
+      const { accessMethods = {} } = byId[collectionId]
+      const { opendap } = accessMethods
+
+      if (!opendap) return
+
+      const { selectedVariables: nextSelectedVariables = [] } = opendap
+
+      if (nextSelectedVariables.length > 0) {
+        selectedVariables[collectionId] = nextSelectedVariables
+      }
+    })
+
+    this.setState({
+      selectedVariables
+    })
+  }
+
+  onPanelClose() {
+    const { onTogglePanels } = this.props
     onTogglePanels(false)
   }
 
-  const onChangePanel = (panelId) => {
+  onChangePanel(panelId) {
+    const { onSetActivePanel } = this.props
     onSetActivePanel(panelId)
   }
 
-  projectIds.forEach((collectionId, index) => {
-    loaded = true
-    const collection = byId[collectionId]
-    const projectCollection = projectById[collectionId]
-    const { metadata } = collection
-    const { dataset_id: title, id, granule_count: granuleCount } = metadata
+  onCheckboxChange(e, variableId, collectionId) {
+    const { selectedVariables, variables } = this.state
+    let newVariables = []
+    const selectedForCollection = selectedVariables[collectionId] || []
+
+    if (variableId === 'all') {
+      if (e.target.checked) {
+        newVariables = [
+          ...selectedForCollection,
+          ...Object.keys(variables)
+        ]
+      } else {
+        // Remove the selected variable if it exists in the selectedForCollection
+        // eslint-disable-next-line arrow-body-style
+        newVariables = selectedForCollection.filter((selectedVariable) => {
```

Comment:
Commented on this below too, we should be able to prevent having to use a disable and just use a multiline arrow body.